### PR TITLE
Feature/options document

### PR DESCRIPTION
**Pull request by CamiloArias12**
# Opciones documentos

## Description
Se ha agregado la opcion de subir documentos  y mejora en algunos componentes

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/modules/files/components/create-document/index.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+
+    const onSubmit = (formData: FormData) => {
+        const data = new FormData()
+        data.set("directoryId", props.curDir.id!.toString());
+        data.set("status", "aprobado")
+        for (const file of selectedFiles) {
+            data.append("files", file)
+        }
+
```
Comment (JosephSC0121) [inline src/modules/files/components/create-document/index.tsx:53]:
```suggestion
```

File: src/modules/files/components/create-document/index.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        console.log(data, formData)
```
Comment (JosephSC0121) [inline src/modules/files/components/create-document/index.tsx:54]:
```suggestion

```

File: src/modules/files/services/actions.ts [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    const response = await getDirectoryService(id);
+    if (response) {
+        return {
+            message: "Institución creada exitosamente",
+            directory: response,
+        };
+    }
+    return {
+        errors: {
+            api: "Error conexion servidor",
```
Comment (JosephSC0121) [inline src/modules/files/services/actions.ts:27]:
```suggestion
            api: "Error en la conexión del servidor",
```

File: src/modules/files/services/actions.ts [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    const response = await getDirectoryChildService(id);
+    if (response) {
+        return {
+            message: "Institución creada exitosamente",
+            directory: response,
+        };
+    }
+    return {
+        errors: {
+            api: "Error conexion servidor",
```
Comment (JosephSC0121) [inline src/modules/files/services/actions.ts:42]:
```suggestion
            api: "Error en la conexión del servidor",
```

**Review by JosephSC0121 — APPROVED** [review]

File: src/modules/common/services/index.tsx [hunk 5]
```diff
@@ -1,89 +1,115 @@
...
+    try {
+        const response = await fetch(`${process.env.API_URL}${path}`, {
+            method: "POST",
+            headers: {
+                Authorization: `Bearer ${session?.user.token}`,
+            },
+            body: body,
+        });
+        console.log(body)
+        console.log(response)
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/common/services/index.tsx:105]: Los console.log borrarlos

File: src/modules/files/components/folder-tree-view/index.tsx [hunk 6]
```diff
@@ -14,169 +14,191 @@ import { getDirectoryAction } from "../../services/actions";
...
-      if (directory.id === value.id && directory.directory) {
-        const directory_node = { ...directoryRoot };
-        handleDirectoryOpenParent(directory_node, value.id);
-        setDirectory(directory_node);
-      }
-    });
-  }, [directories]);
+    useEffect(() => {
+        if (change?.action === "delete-directory" && change.id === directory.id) {
+            console.log(change)
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/files/components/folder-tree-view/index.tsx:61]: Console.log

File: src/app/dashboard/files/[folder]/page.tsx [hunk 7]
```diff
@@ -1,16 +1,22 @@
...
-  const directory = await getDirectoryService(params.folder);
-  const directory_route = await getDirectoryRouteService(params.folder);
-  // TODO: Add logic if there isn't response, or is always get back?
-  return <FolderAll directory={directory!} routeDirectory={directory_route!} />;
+    const directory = await getDirectoryService(params.folder);
+    const directory_route = await getDirectoryRouteService(params.folder);
+    // TODO: Add logic if there isn't response, or is always get back?
+    if (directory || directory_route) {
+        return <FolderAll directory={directory!} routeDirectory={directory_route!} />;
+
```
Comment (Alejandro-Suarez-Rincon) [inline src/app/dashboard/files/[folder]/page.tsx:19]:
```suggestion

```

File: src/modules/common/types/action.ts [hunk 8]
```diff
@@ -1,4 +1,6 @@
 export type CommonActionState = {
-  errors?: {} | null;
-  message?: string | null;
+    errors?: {} | null;
+    message?: string | string[] | null;
 };
+
+
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/common/types/action.ts:6]:
```suggestion
```

File: src/modules/files/components/create-document/index.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+
+    const onSubmit = (formData: FormData) => {
+        const data = new FormData()
+        data.set("directoryId", props.curDir.id!.toString());
+        data.set("status", "aprobado")
+        for (const file of selectedFiles) {
+            data.append("files", file)
+        }
+
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/files/components/create-document/index.tsx:53]:
```suggestion

```

File: src/modules/files/components/folder-document-options/index.tsx [hunk 10]
```diff
@@ -6,84 +6,109 @@ import { Button } from "@/modules/common/components/button";
...
+                />
+            ) : (
+                <FileView layout={props.layout} document={props.value as File} />
+            )}
+
+            <Dropdown
+                showArrow
+                classNames={{
+                    content: " p-0 border rounded-lg max-w-36 min-w-36 "
+
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/files/components/folder-document-options/index.tsx:64]:
```suggestion

```

File: src/modules/files/components/folder-document-options/index.tsx [hunk 11]
```diff
@@ -6,84 +6,109 @@ import { Button } from "@/modules/common/components/button";
...
+
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/files/components/folder-document-options/index.tsx:65]:
```suggestion

```

**Pull request merged**